core/src/ipc.rs

```diff
@@ -131,3 +135,99 @@ impl<T> FromRawFd for IpcReceiver<T> {
         }
     }
 }
+
+/// Creates a pair of sockets that are ment for passing file descriptors to IO resources of type T.
+pub fn io_pair<T>() -> TypedResult<(IoSender<T>, IoReceiver<T>)>
+where
+    T: AsRawFd,
+{
+    let (tx, rx) = socketpair(
+        AddressFamily::Unix,
+        SockType::Datagram,
+        None,
+        SockFlag::SOCK_NONBLOCK,
```

Comment:
I would at least document the non-blocking property in the functions description, as non-blocking behavior is not the default.